### DISCUSSION TITLE
How to merge a merge module of x64 to a msi module of ARM64.

### DISCUSSION THREAD

**Discussion by taka-ka** [wixtoolset/issues#6796] [Questions] [answered]
I would like to a merge a merge module of x64 to a msi module of ARM64.
But a error occures with the following message.

>xxx.msm' is a 64-bit merge module but the product consuming it is 32-bit. 32-bit products can consume only 32-bit merge modules.

It seems that when the MergeEx function called, the error type of msmErrorPlatformMismatch was returned.
I was trying to register the latest mergemod.dll (5\.0.22000.194) , but the result was same.

**Comment by robmen** [comment, marked as answer]
The exact error message is probably a little dated but I think it's pointing in the right direction. When a Merge Module declares its supported platform, the merge tooling will not merge it into a mismatched platform. In this case, the error message should probably say `The MSM is x64 but the product is ARM64. x64 merge modules can only be merged into x64 products.` It's possible that Microsoft needs to update the Merge API to add ARM64 into its support matrix.
